packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.types.ts

```diff
@@ -0,0 +1,213 @@
+   * @default 100%
+   */
+  widthInPercentage?: number;
+
+  /**
+   * Sets the width of the shimmer wave wrapper to an exact value in pixels.
+   * If none of the widths provided, it defaults to 100%.
+   */
+  widthInPixel?: number;
+
+  /**
+   * Controls when the shimmer is swapped with actual data through an animated transition.
+   * @default false
+   */
+  isDataLoaded?: boolean;
+
+  /**
+   * Elements to render in one line of the Shimmer.
+   */
+  shimmerElements?: IShimmerElement[];
+
+  /**
+   * Custom elements when necessary to build complex placeholder skeletons.
+   */
+  customElementsGroup?: React.ReactNode;
+
+  /**
+   * Localized string of the status label for screen reader
+   */
+  ariaLabel?: string;
+
+  /**
+   * Call to provide customized styling that will layer on top of the variant rules.
+   */
+  styles?: IStyleFunctionOrObject<IShimmerStyleProps, IShimmerStyles>;
+
+  /**
+   * Additional CSS class(es) to apply to the Shimmer container.
+   */
+  className?: string;
+
+  /**
+   * Theme provided by High-Order Component.
+   */
+  theme?: ITheme;
+}
+
+/**
+ * Shimmer Elements Interface
+ */
+export interface IShimmerElement {
+  /**
+   * Required for every element you intend to use.
+   */
+  type: ShimmerElementType;
+
+  /**
+   * The height of the element (ICircle, ILine, IGap) in pixels.
+   * Read more details for each specific element.
+   */
+  height?: number;
+
+  /**
+   * The width of the element (ILine, IGap) in pixels.
+   * Read more details for each specific element.
+   */
+  widthInPixel?: number;
+
+  /**
+   * The width of the element (ILine, IGap) in pixels.
+   * Read more details for each specific element.
+   */
+  widthInPercentage?: number;
+
+  /**
+   * The vertical alignemt of the element (ICircle, ILine).
+   * @default center
+   */
+  verticalAlign?: ShimmerElementVerticalAlign;
+}
+
+export interface ILine extends IShimmerElement {
+  /**
+   * Sets the height of the shimmer line in pixels.
+   * @default 16px
+   */
+  height?: number;
+
+  /**
+   * The value provided will represent the width as '%' relative to shimmer wrapper.
+   * @default 100%
+   */
+  widthInPercentage?: number;
+
+  /**
+   * Sets the width of the Line to an exact value in pixels.
+   * If none of the widths provided, it defaults to 100%.
+   */
+  widthInPixel?: number;
+}
+
+export interface ICircle extends IShimmerElement {
+  /**
+   * Sets the height of the shimmer circle in pixels.
+   * Minimum supported 10px.
+   * @default 24px
+   */
+  height?: number;
+}
+
+export interface IGap extends IShimmerElement {
+  /**
+   * Sets the height of the shimmer gap in pixels.
+   * @default 16px
+   */
+  height?: number;
+  /**
+   * The value provided will represent the width as '%' relative to shimmer wrapper.
+   * If none of the widths provided, it defaults to 10px.
+   */
+  widthInPercentage?: number;
```

Comment:
this is all so confusing. Why not just have width?

Reply:
Fixed